src/Streams.sol

```diff
@@ -811,19 +859,14 @@ abstract contract Streams {
             return uint32(notEnoughEnd);
         }
 
-        if (hint1 > enoughEnd && hint1 < notEnoughEnd) {
-            if (_isBalanceEnough(balance, configs, configsLen, hint1)) {
-                enoughEnd = hint1;
-            } else {
-                notEnoughEnd = hint1;
-            }
-        }
-
-        if (hint2 > enoughEnd && hint2 < notEnoughEnd) {
-            if (_isBalanceEnough(balance, configs, configsLen, hint2)) {
-                enoughEnd = hint2;
+        while (maxEndHints.hasHints()) {
```

Comment:
Just to double check.
```solidity
uint256 enoughEnd = _currTimestamp();
```
Current timestamp will be always `enoughEnd`. This is an invariant of the system and can never be false.